monome/teletype
# Proposal: separate scene serialization logic from USB disk logic, add scene serialization tests

### What does this PR do?

This PR separates the scene parsing code and the USB disk handling code, so scenes could theoretically be loaded from and saved to other sources besides the USB drive. 

It creates an abstract interface for scene serialization and leverages it to add unit tests for scene I/O.  When invoked in the USB disk mode, the sequence of operations is more or less unchanged; this doesn't substantially alter the user-visible behavior.

The included presets are also updated to the latest syntax (newer II ops, PRE colon spacing) so tests can round-trip them and compare the output against the original.

This also includes fixes for two bugs discovered while doing this work:
   * inits scene calibration data to reasonable values to avoid divide-by-zero in `process_tests` on some x86 hardware.
   * enlarges the text string buffer used for `WRITE.../READ...` in USB disk mode, avoiding overflowing the buffer and potentially smashing neighboring stack vars.


### How should this be manually tested?

* Load a USB stick with a selection of scenes in `tt*.txt`
* Insert USB stick into teletype, confirm WRITE/READ cycle completes
* Remove, attach keyboard, confirm scenes are loaded correctly
* Re-insert USB stick, repeat WRITE/READ cycle
* Insert USB stick into computer, compare saved files to originals with something like the below command; files should be the same 
   `for f in tt*.txt; do git diff $f ${f%.*}s.txt; done`

### Any background context you want to provide?

The original motivation for this was to be able to use the scene parser in VCV Rack, but also with an eye to benefit hardware development down the road. Hopefully this makes it a little easier to contemplate improvements like:
* CRLF line ending support
* #104 
* scene loading over i2c?
* new USB disk management UI??

### I have,
* ~~updated the documentation~~
* ~~updated `help_mode.c` (if applicable)~~
